activerecord/lib/active_record/relation.rb

```diff
@@ -15,6 +15,10 @@ class Relation
 
     VALUE_METHODS = MULTI_VALUE_METHODS + SINGLE_VALUE_METHODS + CLAUSE_METHODS
 
+    EXCLUDED_OR_METHODS = [:extending, :where, :having, :unscope, :references]
+
+    STRUCTURAL_OR_METHODS = VALUE_METHODS - EXCLUDED_OR_METHODS
```

Comment:
Can you move those constants back? They don't need to be in relation since they are only used in one place.

Reply:
Firstly, I thought of creating a function as logic will be in function scope only, but maybe creating a variable inside the function would be a better idea. Also about the `EXCLUDED_OR_METHODS` block, it should be attached to Relation class as it defines the column which shouldn't be included while doing `OR`. What are your views @rafaelfranca ?

Reply:
My views is that the code as it was before for this constant is better. It is isolated, and only exists because we didn't want to allocate extra arrays when calling `structurally_incompatible_values_for_or`, so it is better to continue to be closer to that method, and there is no need to create the `EXCLUDED_OR_METHODS` constant as well.

Reply:
Done as suggested